src/connection/xmpp/handlers/headlineMessage.ts

```diff
@@ -8,6 +8,13 @@ export default class extends AbstractHandler {
       let fromJid = new JID(stanza.getAttribute('from'));
       let connection = this.account.getConnection();
       let myJid = connection.getJID();
+      let nicknameObject = this.account.getContact(myJid).getNicknameObject();
+      let nickname: string = nicknameObject.getNickFromStanza(stanza);
+      let oldName: string = nicknameObject.getNickname();
+
+      if ((nickname !== undefined) && (oldName !== nickname)) {
+         this.account.getContact(fromJid).getNicknameObject().setContactNickname(nickname);
+      }
```

Comment:
Stay DRY. See `chatMessage.ts`.

Reply:
How would we reuse the code from chatMessage for this if statement?